backend/src/club/club.gateway.ts

```diff
@@ -0,0 +1,39 @@
+import {
+  SubscribeMessage, WebSocketGateway,
+  ConnectedSocket, WsResponse,
+  OnGatewayConnection, OnGatewayDisconnect,
+} from '@nestjs/websockets';
+import { Socket } from 'socket.io';
+import { Logger, UseGuards } from '@nestjs/common';
+import { RolesGuard } from '../roles/roles.guard';
+import { AllowedRoles } from '../roles/roles.decorator';
+import { Roles } from '../roles/roles.enum';
+import * as gatewayEvents from './gateway-events.json';
+import { WsJwtGuard } from '../authentication/ws.jwt.guard';
+
+@WebSocketGateway({ transports: ['polling'], origins: '*:*' })
+export class ClubGateway implements OnGatewayConnection, OnGatewayDisconnect {
+
+  private readonly logger = new Logger('Club Gateway');
+  connectedJuniors: Socket[] = [];
+
+  handleConnection(client: Socket) {
+    this.logger.log(`connected ${client.id}`);
+  }
+
+  handleDisconnect(client: Socket) {
+    this.connectedJuniors = this.connectedJuniors.filter(c => c.id !== client.id);
+    this.logger.log(`diconnected ${client.id}`);
+  }
+
+  @UseGuards(WsJwtGuard, RolesGuard)
+  @AllowedRoles(Roles.JUNIOR)
+  @SubscribeMessage(gatewayEvents.checkIn)
```

Comment:
Should we use some other event for establishing the connection? That way it would be easier for frontend to handle the actual check-in event.

Reply:
The initial connection gets established during "handleConnection()". The method in question is currently used to a). Inform the fronend that they have successfully connected, b). Allows the backend to associate an id with the client to be used when a check-in has occurred. 
Currently I feel this is a simple approach to ensure that only correct users are notified. I am however open to alternative solutions.